ddl/util/syncer_test.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/terror"
+	. "github.com/pingcap/tidb/ddl"
+	. "github.com/pingcap/tidb/ddl/util"
+	"github.com/pingcap/tidb/owner"
+	"github.com/pingcap/tidb/store/mockstore"
+	goctx "golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+const minInterval = 10 * time.Nanosecond // It's used to test timeout.
+
+func TestSyncerSimple(t *testing.T) {
+	testLease := 5 * time.Millisecond
+	origin := CheckVersFirstWaitTime
+	CheckVersFirstWaitTime = 0
+	defer func() {
+		CheckVersFirstWaitTime = origin
+	}()
+
+	store, err := mockstore.NewMockTikvStore()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer store.Close()
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+	cli := clus.RandClient()
+	ctx := goctx.Background()
+	d := NewDDL(ctx, cli, store, nil, nil, testLease, nil)
+	defer d.Stop()
+
+	// for init function
+	if err = d.SchemaSyncer().Init(ctx); err != nil {
+		t.Fatalf("schema version syncer init failed %v", err)
+	}
+	resp, err := cli.Get(ctx, DDLAllSchemaVersions, clientv3.WithPrefix())
+	if err != nil {
+		t.Fatalf("client get version failed %v", err)
+	}
+	key := DDLAllSchemaVersions + "/" + d.OwnerManager().ID()
+	checkRespKV(t, 1, key, InitialVersion, resp.Kvs...)
+	// for MustGetGlobalVersion function
+	globalVer, err := d.SchemaSyncer().MustGetGlobalVersion(ctx)
+	if err != nil {
+		t.Fatalf("client get global version failed %v", err)
+	}
+	if InitialVersion != fmt.Sprintf("%d", globalVer) {
+		t.Fatalf("client get global version %d isn't equal to init version %s", globalVer, InitialVersion)
+	}
+	childCtx, _ := goctx.WithTimeout(ctx, minInterval)
+	_, err = d.SchemaSyncer().MustGetGlobalVersion(childCtx)
+	if !isTimeoutError(err) {
+		t.Fatalf("client get global version result not match, err %v", err)
+	}
+
+	d1 := NewDDL(ctx, cli, store, nil, nil, testLease, nil)
+	defer d1.Stop()
+	if err = d1.SchemaSyncer().Init(ctx); err != nil {
+		t.Fatalf("schema version syncer init failed %v", err)
+	}
+
+	// for watchCh
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	currentVer := int64(123)
+	go func() {
+		defer wg.Done()
+		select {
+		case resp := <-d.SchemaSyncer().GlobalVersionCh():
+			if len(resp.Events) < 1 {
+				t.Fatalf("get chan events count less than 1")
+			}
+			checkRespKV(t, 1, DDLGlobalSchemaVersion, fmt.Sprintf("%v", currentVer), resp.Events[0].Kv)
+		case <-time.After(100 * time.Millisecond):
+			t.Fatalf("get udpate version failed")
+		}
+	}()
+
+	// for update latestSchemaVersion
+	err = d.SchemaSyncer().OwnerUpdateGlobalVersion(ctx, currentVer)
+	if err != nil {
+		t.Fatalf("update latest schema version failed %v", err)
+	}
+
+	wg.Wait()
+
+	// for CheckAllVersions
+	childCtx, cancel := goctx.WithTimeout(ctx, 20*time.Millisecond)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
+	if err == nil {
+		t.Fatalf("check result not match")
```

Comment:
Please use assert to check whether the err is notNil

Reply:
This test needs the function of `NewClusterV3` that needs the argument of `testing.T`. So I don't use assert.